Main CI fails on GridView AgentTerminalSlot prop mismatch

## Problem
The `tests` workflow on `main` fails during `npm run lint` after merging #587.

Failure:

```text
src/views/GridView.tsx(402,21): error TS2322: Type '{ sessionId: string; provider: string | undefined; isMaximized: boolean; isSelected: boolean; theme: "system" | "dark" | "light"; workspacePath: string; onTerminalFocus: ((agentId: string) => void) | undefined; onTitleChange: (id: string, title: string) => void; }' is not assignable to type 'IntrinsicAttributes & AgentTerminalSlotProps'.
  Property 'isSelected' does not exist on type 'IntrinsicAttributes & AgentTerminalSlotProps'.
```

Run: https://github.com/wardian-app/Wardian/actions/runs/28150723971

## Expected
`main` should compile cleanly and preserve Grid terminal selection behavior.

--- a/src/views/GridView.test.tsx
+++ b/src/views/GridView.test.tsx
@@ -13,9 +13,10 @@ vi.mock('../features/terminal/AgentTerminal', async () => {
   return {
     AgentTerminal: React.memo((props: {
       sessionId: string;
+      isSelected?: boolean;
       onTerminalFocus?: () => void;
     }) => {
-      terminalRenderSpy(props.sessionId);
+      terminalRenderSpy(props);
       return React.createElement(
         'div',
         {
@@ -155,6 +156,21 @@ describe('GridView maximize behavior', () => {
     screen.getByTestId('terminal-agent-2').focus();
 
     expect(onTerminalFocus).toHaveBeenCalledWith('agent-2');
+  });
+
+  it('passes grid selection state through to terminal cards', () => {
+    renderGrid(null, agents, vi.fn(), {
+      selectedAgentIds: new Set(['agent-1']),
+    });
+
+    expect(terminalRenderSpy).toHaveBeenCalledWith(expect.objectContaining({
+      sessionId: 'agent-1',
+      isSelected: true,
+    }));
+    expect(terminalRenderSpy).toHaveBeenCalledWith(expect.objectContaining({
+      sessionId: 'agent-2',
+      isSelected: false,
+    }));
   });
 
   it('does not size each mobile card to the full viewport height', () => {

--- a/src/views/GridView.tsx
+++ b/src/views/GridView.tsx
@@ -68,6 +68,7 @@ interface AgentTerminalSlotProps {
   sessionId: string;
   provider?: string;
   isMaximized: boolean;
+  isSelected: boolean;
   theme: "dark" | "light" | "system";
   workspacePath?: string;
   onTerminalFocus?: (agentId: string) => void;
@@ -78,6 +79,7 @@ const AgentTerminalSlot = React.memo(function AgentTerminalSlot({
   sessionId,
   provider,
   isMaximized,
+  isSelected,
   theme,
   workspacePath,
   onTerminalFocus,
@@ -96,6 +98,7 @@ const AgentTerminalSlot = React.memo(function AgentTerminalSlot({
       sessionId={sessionId}
       provider={provider}
       isMaximized={isMaximized}
+      isSelected={isSelected}
       theme={theme}
       workspacePath={workspacePath}
       onTerminalFocus={handleTerminalFocus}

